messaging/src/main/java/org/springframework/security/messaging/access/expression/MessageExpressionConfigAttribute.java

```diff
@@ -15,30 +15,48 @@
  */
 package org.springframework.security.messaging.access.expression;
 
+import org.springframework.expression.EvaluationContext;
 import org.springframework.expression.Expression;
 import org.springframework.messaging.Message;
 import org.springframework.security.access.ConfigAttribute;
+import org.springframework.security.messaging.util.matcher.MessageMatcher;
+import org.springframework.security.messaging.util.matcher.SimpDestinationMessageMatcher;
 import org.springframework.util.Assert;
 
+import java.util.Map;
+
 /**
  * Simple expression configuration attribute for use in {@link Message} authorizations.
  *
  * @since 4.0
  * @author Rob Winch
+ * @author Daniel Bustamante Ospina
  */
 @SuppressWarnings("serial")
-class MessageExpressionConfigAttribute implements ConfigAttribute {
+class MessageExpressionConfigAttribute implements ConfigAttribute, EvaluationContextPostProcessor<Message<?>> {
 	private final Expression authorizeExpression;
+	private final MessageMatcher<?> matcher;
+
 
 	/**
 	 * Creates a new instance
 	 *
 	 * @param authorizeExpression the {@link Expression} to use. Cannot be null
+	 * @param matcher the {@link MessageMatcher} used to match the messages.
 	 */
-	public MessageExpressionConfigAttribute(Expression authorizeExpression) {
+	public MessageExpressionConfigAttribute(Expression authorizeExpression, MessageMatcher<?> matcher) {
 		Assert.notNull(authorizeExpression, "authorizeExpression cannot be null");
-
 		this.authorizeExpression = authorizeExpression;
+		this.matcher = matcher;
+	}
+
+	/**
+	 * Creates a new instance
+	 *
+	 * @param authorizeExpression the {@link Expression} to use. Cannot be null
+	 */
+	public MessageExpressionConfigAttribute(Expression authorizeExpression) {
```

Comment:
Since this is a package private class we don't need to have this additional constructor. We should remove it because other wise there is the chance that the matcher is null which means we need to do null checks within the code. If we validate up front this is non-null then we no longer need null checks